validator/server_api/validation_client.go

```diff
@@ -0,0 +1,214 @@
+package server_api
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+
+	"github.com/offchainlabs/nitro/validator"
+
+	"github.com/offchainlabs/nitro/util/containers"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+
+	"github.com/offchainlabs/nitro/validator/server_common"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type ValidationClient struct {
+	stopwaiter.StopWaiter
+	client    *rpc.Client
+	url       string
+	name      string
+	jwtSecret []byte
+}
+
+func NewValidationClient(url string, jwtSecret []byte) *ValidationClient {
+	return &ValidationClient{
+		url:       url,
+		jwtSecret: jwtSecret,
+	}
+}
+
+func (c *ValidationClient) Launch(entry *validator.ValidationInput, moduleRoot common.Hash) validator.ValidationRun {
+	valrun := server_common.NewValRun(moduleRoot)
+	c.LaunchThread(func(ctx context.Context) {
```

Comment:
I'm thinking about whether or not Launch should have its own context. Since we shutdown the validation client before waiting for any pending validations, there isn't a huge impact, but I think on reorgs we cancel running validations? If we want to change this it'd probably be better to do so in a follow-up PR.

If we do add a context to Launch, we should also add it to the other ValidationClient and ExecutionClient methods.

Reply:
ValidationRun is not attached to context, but it does have a Close method that can be called.
It's similar, but I think this interface would be a better fit for remote execution. E.g when connecting to multiple validation servers, if one server dropped the underlying validation run could launch itself on a different server.
Same could be done with context, of course..